src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -366,9 +373,6 @@ public boolean apply(@Nullable Container container) {
      *
      */
     private synchronized boolean addProvisionedSlave(String ami, int amiCap) throws Exception {
-        if( amiCap == 0 )
-            return true;
-
```

Comment:
In which case it should be `return false;`

This leads to another issue:
In the event an ami has too many instances, the plugin will not look for other images that also satisfy the conditions. When matching labels, it should really be getting a list of templates that are compatible.